src/Calculator/GraphingSettings.xaml.cpp

```diff
@@ -0,0 +1,220 @@
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Navigation;
+
+GraphingSettings::GraphingSettings()
+{
+    InitializeComponent();
+
+    auto resLoader = AppResourceProvider::GetInstance();
+
+    x_MinHeading->Text = resLoader.GetResourceString(L"x-min");
+    x_MaxHeading->Text = resLoader.GetResourceString(L"x-max");
+    y_MinHeading->Text = resLoader.GetResourceString(L"y-min");
+    y_MaxHeading->Text = resLoader.GetResourceString(L"y-max");
+
+    GridHeading->Text = resLoader.GetResourceString(L"GridHeading");
+    UnitsHeading->Text = resLoader.GetResourceString(L"UnitsHeading");
+
+    Radians->Content = resLoader.GetResourceString(L"TrigModeRadians");
+    Degrees->Content = resLoader.GetResourceString(L"TrigModeDegrees");
+    Gradians->Content = resLoader.GetResourceString(L"TrigModeGradians");
+}
+
+void GraphingSettings::GraphingCalculator_DataContextChanged(FrameworkElement ^ sender, DataContextChangedEventArgs ^ args)
+{
+    // The graph control we are working with
+    m_ParentGrapher = (GraphControl::Grapher ^) args->NewValue;
+}
+
+void GraphingSettings::TrigUnitModeClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    if (Degrees->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Degrees;
+    }
+
+    if (Radians->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Radians;
+    }
+
+    if (Gradians->IsChecked->Value == true)
+    {
+        m_ParentGrapher->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Grads;
+    }
+}
+
+void GraphingSettings::UpdateExtents(Platform::Object ^ sender)
+{
+    FrameworkElement ^ fe = (FrameworkElement ^) sender;
+    FrameworkElement ^ currentFocus = (FrameworkElement ^) FocusManager::GetFocusedElement();
+
+    bool thisControlHasFocus = (fe == currentFocus);
+    TextBox ^ tb = reinterpret_cast<TextBox ^>(sender);
+    bool widthChanged = false;
+    bool heightChanged = false;
+
+    try
+    {
+        double newVal = std::stod(tb->Text->Begin());
+        if (tb->Name == "x_min" && newVal != m_xMin)
+        {
+            widthChanged = true;
+            m_xMin = newVal;
+            m_ParentGrapher->XAxisMin = m_xMin;
+        }
+
+        if (tb->Name == "x_max" && newVal != m_xMax)
+        {
+            widthChanged = true;
+            m_xMax = newVal;
+            m_ParentGrapher->XAxisMax = m_xMax;
+        }
+
+        if (tb->Name == "y_min" && newVal != m_yMin)
+        {
+            heightChanged = true;
+            m_yMin = newVal;
+            m_ParentGrapher->YAxisMin = m_yMin;
+        }
+
+        if (tb->Name == "y_max" && newVal != m_yMax)
+        {
+            heightChanged = true;
+            m_yMax = newVal;
+            m_ParentGrapher->YAxisMax = m_yMax;
+        }
+
+        // Only the focused element get's to trigger the recompute
+
+        if (thisControlHasFocus)
```

Comment:
Is this check necessary? If we only call this from OnLosingFocus and OnKeyDown then I think it should only fire once for the specific control that fired the event, unless OnKeyDown fires for all controls even if they aren't the one having text inputted?